### PR TITLE
DROOLS-3805 : All line separators in RHS in spreadsheet are escaped

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-3805

### CODE REVIEW

File: drools-decisiontables/src/main/java/org/drools/decisiontable/parser/RhsBuilder.java [hunk 1]
```diff
@@ -87,12 +87,10 @@ public void clearValues() {
...
         StringBuffer buf = new StringBuffer();
-        for (Iterator<String> iter = this.values.iterator(); iter.hasNext(); ) {
-            String next = iter.next();
-            String replace = next.replace("\n", "\\n");
-            buf.append(replace);
+        for ( Iterator<String> iter = this.values.iterator(); iter.hasNext(); ) {
+            buf.append( iter.next() );
             if (iter.hasNext()) {
-                buf.append('\n');
+                buf.append( '\n' );
```
Comment (Rikkola) [inline drools-decisiontables/src/main/java/org/drools/decisiontable/parser/RhsBuilder.java:93]: Reverting the previous "fix".

File: drools-templates/src/main/java/org/drools/template/model/SnippetBuilder.java [hunk 2]
```diff
@@ -179,7 +179,7 @@ private String replace( final String text,
         while ( ( end = text.indexOf( repl,
                                       start ) ) != -1 ) {
             buf.append( text.substring( start,
-                                        end ) ).append( with );
+                                        end ) ).append( with.replace("\n", "\\n") );
```
Comment (Rikkola) [inline drools-templates/src/main/java/org/drools/template/model/SnippetBuilder.java:182]: Just replacing for cells that are added in.

**Comment by Rikkola** [comment]
@jomarko @manstis And here one too, I hope this would be the last, but making at least one more this week ;)

File: drools-decisiontables/src/test/java/org/drools/decisiontable/MakeSureMultiLinesWorkTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package org.drools.decisiontable;
```
Comment (manstis) [inline drools-decisiontables/src/test/java/org/drools/decisiontable/MakeSureMultiLinesWorkTest.java:15]: License please.

Comment (Rikkola) [reply drools-decisiontables/src/test/java/org/drools/decisiontable/MakeSureMultiLinesWorkTest.java:15]: Added.

**Review by manstis — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Review by jomarko — COMMENTED** [review]
@Rikkola I have no comments to code. Just not sure how it should behave if multiline text is as string constant. See https://github.com/Rikkola/drools/pull/2

**Comment by jomarko** [comment]
We can not forget to merge this https://github.com/kiegroup/drools/pull/2293

**Review by jomarko — APPROVED** [review]
@Rikkola thank you!

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers This can be merged thank you.

**Pull request merged**